sematic/ui/src/index.tsx

```diff
@@ -183,6 +185,49 @@ function Router() {
   return <RouterProvider router={router} />;
 }
 
+(async () => {
+  const currentUrl = new URL(window.location.toString());
+  const currentHostName = await sha1(currentUrl.hostname);
+
+  posthog.init( 
+    'phc_nJlFf7MpsrzF5pPaSEQi5GKyTSDjHRcqqL808VLRNXc', { 
+      api_host: 'https://app.posthog.com',
+      autocapture: false,
+      debug: true,
+      disable_session_recording: true,
+      capture_pageview: false,
+      capture_pageleave: false,
+      persistence: "localStorage",
+      property_blacklist: [
+        '$referrer', '$referring_domain', '$initial_current_url', '$initial_referrer',
+        '$initial_referring_domain'
+      ],
+      loaded: () => {
+        setupPostHogOptout();
+        posthog.capture('$pageview');
+      },
+
+      sanitize_properties: (properties: Properties, event_name: string) => {
+        const currentUrl = new URL(window.location.toString());
+
+        // Use obfuscated host name
+        currentUrl.hostname = currentHostName;
+
+        if ('$current_url' in properties) {
+          properties['$current_url'] = currentUrl.toString();
+        }
+
+        if ('$host' in properties) {
+          properties['$host'] = currentUrl.host;
```

Comment:
Why collect the host?
If you updated the hostname, is the host updated as well?

Reply:
Good question, 

Optionally we can blacklist the `host` property from being sent. But keeping it is also useful because we can rely on it to filter or segment the collected events by `host` without the complex logic to extract the host from the URL (it might not be even supported to use a formula in the event platform).

It appears when this callback is called, PostHog has already collected the values for '$current_url' and '$host'. So changing the current URL location would not impact them, unless explicitly overriding them.

Reply:
My understanding is that the host contains the hostname + port. So if we only obfuscate the hostname, then the info will still be available in the host, defeating the purpose of the obfuscation. Am I missing something?

Reply:
My bad, sorry, I totally missed the point:

Yes, the host will be updated when the hostname is updated if the `URL` data structure is being used.

<img width="309" alt="image" src="https://user-images.githubusercontent.com/1046489/219799037-f4bd0993-0ce6-4e3c-a0d8-a3c2135172e5.png">